### ISSUE TITLE
Docker PullImage API is broken

### ISSUE THREAD

**Issue by diptanu** [samalba/dockerclient#66]
After the PR around better error message was merged into master, the PullImage API seems to be broken.

I am getting the following error message -
```
invalid character '{' after top-level value
```
when I am downloading an image from a private repository, for example -  myregistry/busybox

My fork which was created before the PR #49 was merged works as expected - https://github.com/diptanu/dockerclient

@donhcd @vieux @samalba

**Comment by donhcd** [comment]
you're right - I assumed that the JSON objects would be newline-separated but that's unfortunately not true. I'm working on a PR to fix this

**Comment by donhcd** [comment]
@diptanu please take a look and see if this works for you ^  Parsing all of the json objects is pretty negligible compared to the network io, so I don't think there's anything wrong with me doing that.

**Comment by vieux** [comment]
I merged #66 because it fixed the issue for me. 

@diptanu please confirm
